internal/incident/incident.go

```diff
@@ -533,6 +479,64 @@ func (i *Incident) RestoreEscalationStateRules(states []*EscalationState) {
 	}
 }
 
+// getEscalationRecipientsChannel returns the configured channel of all the provided escalation recipients.
+func (i *Incident) getEscalationRecipientsChannel(escalation *rule.Escalation, channels recipientsChannelType, t time.Time) {
+	managed := i.HasManager()
+	for _, escalationRecipient := range escalation.Recipients {
+		state := i.Recipients[escalationRecipient.Key]
+		if state == nil {
+			continue
+		}
+
+		if !managed || state.Role > RoleRecipient {
+			for _, c := range escalationRecipient.Recipient.GetContactsAt(t) {
+				if channels[c] == nil {
+					channels[c] = make(map[string]struct{})
+				}
+				if escalationRecipient.ChannelType.Valid {
+					channels[c][escalationRecipient.ChannelType.String] = struct{}{}
+				} else {
+					channels[c][c.DefaultChannel] = struct{}{}
+				}
+			}
+		}
+	}
+}
+
+// getRecipientsChannel returns all the configured channels of the current incident and escalation recipients.
+func (i *Incident) getRecipientsChannel(t time.Time) recipientsChannelType {
+	managed := i.HasManager()
+	contactChannels := make(recipientsChannelType)
+	for escalationID := range i.EscalationState {
+		escalation := i.runtimeConfig.GetRuleEscalation(escalationID)
+		i.getEscalationRecipientsChannel(escalation, contactChannels, t)
+	}
+
+	escalationRecipients := make(map[recipient.Key]bool)
+	for contact := range contactChannels {
+		escalationRecipients[recipient.Key{ContactID: utils.ToDBInt(contact.ID)}] = true
+	}
+
+	for recipientKey, state := range i.Recipients {
+		r := i.runtimeConfig.GetRecipient(recipientKey)
+		if r == nil {
+			continue
+		}
+
+		isEscalationRecipient := escalationRecipients[recipientKey]
+		if !isEscalationRecipient && (!managed || state.Role > RoleRecipient) {
+			for _, contact := range r.GetContactsAt(t) {
+				if contactChannels[contact] == nil {
+					contactChannels[contact] = make(map[string]struct{})
+				}
+				contactChannels[contact][contact.DefaultChannel] = struct{}{}
+			}
+		}
+	}
+
+	return contactChannels
+}
```

Comment:
Please add a few comments within the function that say what the individual parts do.